python/pyarrow/compute.py

```diff
@@ -253,6 +255,68 @@ def match_substring(array, pattern):
                          MatchSubstringOptions(pattern))
 
 
+def split_pattern(array, pattern, max_splits=-1, reverse=False):
```

Comment:
You don't need this, it will be exposed automatically. However, you'll need to expand `_option_classes`.

Reply:
I see that changed in #8163, nice work. But that would make the API less friendly, having to pass in the options, instead of having an argument with default parameters. If you are strongly in favor of this, I'm ok with it, I'm not sure how 'user friendly' the arrow API needs to be.

Reply:
No, it should work with direct keyword arguments. See https://github.com/apache/arrow/blob/master/python/pyarrow/tests/test_compute.py#L233-L260 for a test.

Reply:
It's true that the docstring is a bit suboptimal still. Hopefully we can improve that in the future:
```
>>> pc.min_max?
Signature: pc.min_max(arg, *, options=None, memory_pool=None, **kwargs)
Docstring:
Call compute function 'min_max' with the given argument.

Parameters
----------
arg : Array-like or scalar-like
    Argument to compute function
memory_pool : pyarrow.MemoryPool, optional
    If not passed, will allocate memory from the default memory pool.
options : pyarrow.compute.MinMaxOptions, optional
    Parameters altering compute function semantics
**kwargs: optional
    Parameters for MinMaxOptions constructor.  Either `options`
    or `**kwargs` can be passed, but not both at the same time.
```